cmake/Modules/HandleCUDAComputeCapability.cmake

```diff
@@ -8,24 +8,55 @@
 # those without the CUDA_ prefix
 
 if (NOT CUDA_TARGET_COMPUTE_CAPABILITY)
-	if("$ENV{CUDA_SM}" STREQUAL "")
-		set(ENV{CUDA_INCLUDE_DIRS} "${CUDA_INCLUDE_DIRS}")
-		set(ENV{CUDA_CUDART_LIBRARY} "${CUDA_CUDART_LIBRARY}")
-		set(ENV{CMAKE_CXX_COMPILER} "${CMAKE_CXX_COMPILER}")
-		execute_process(COMMAND bash -c "${CMAKE_CURRENT_SOURCE_DIR}/scripts/get_cuda_sm.sh"  OUTPUT_VARIABLE CUDA_TARGET_COMPUTE_CAPABILITY_) 
-	else()
-		set(CUDA_TARGET_COMPUTE_CAPABILITY_ $ENV{CUDA_SM})
+	if (CMAKE_CUDA_COMPILER_LOADED)
+		set(CUDAFILE ${CMAKE_CURRENT_BINARY_DIR}/check_cuda.cu)
```

Comment:
I'm not sure I like this location for the file - that's not really a place to put transient files in. Can you get it into `CMakeFiles/` somehow?

Reply:
${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY} should to the trick?

Reply:
Sure, but let's call it something like `query_gpu_sm_value.cu` or `query_gpu_compute_capability.cu`.